### ISSUE TITLE
Add magic __isset to classes with __get

### ISSUE THREAD

**Issue by MGatner** [codeigniter4/CodeIgniter4#2219]
The following classes have a magic `__get()` method without a corresponding `__isset()`:
* CodeIgniter4/system/Database/BaseConnection.php
* CodeIgniter4/system/Model.php
* CodeIgniter4/system/Session/Session.php
* CodeIgniter4/system/I18n/Time.php
* CodeIgniter4/system/CLI/BaseCommand.php
* CodeIgniter4/system/I18n/TimeDifference.php
* CodeIgniter4/system/Encryption/Encryption.php
* CodeIgniter4/system/Encryption/Handlers/BaseHandler.php

In general this is a bad idea because some functions that call language constructs like `isset()` on an instance will fail to return `true` when `__get()` would still return a valid item. See https://github.com/codeigniter4/CodeIgniter4/pull/2205 for one example.

Unless there's some compelling reason not to, I propose each of these classes should have a basic `__isset()` added, modeled after their `__get()`. E.g.:
```
    public function __isset($name)
    {
        return isset($this->attributes[$name]);
    }
```

**Comment by lonnieezell** [comment]
No compelling reason not to do that.

**Comment by jim-parry** [comment]
Fixed by #2231
